libraries/abstractions/common_io/test/test_iot_i2c.c

```diff
@@ -871,7 +867,7 @@ TEST( TEST_IOT_I2C, AFQP_IotI2CWriteAsyncFailIoctl )
 {
     IotI2CHandle_t xI2CHandle;
     int32_t lRetVal;
-    uint8_t writeVal[] = { xtestIotI2CDeviceRegister, uctestIotI2CWriteVal, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF };
+    uint8_t writeVal[ IOT_TEST_COMMON_IO_I2C_MAX_WRITE_LENGTH ] = { xtestIotI2CDeviceRegister, uctestIotI2CWriteVal };
```

Comment:
Is one byte too few here? Write may finish before the next api call especially in the multi-task env.

Reply:
IOT_TEST_COMMON_IO_I2C_MAX_WRITE_LENGTH is configurable. So the writeVal array is actually initialize to the size of what you want. The first two bytes are xtestIotI2CDeviceRegister and uctestIotI2CWriteVal, and the rest of writeVal will be filled 0. I think 0 is fine because the value doesn't matter

Reply:
The problem is user may neglect to set more bytes and get wrong result if it is not enforced here.

Reply:
I see your concern. I plan to give a default value to IOT_TEST_COMMON_IO_I2C_MAX_WRITE_LENGTH and vendor can override.

Reply:
Is it possible to give something more than 2? Will it affect TI?

Reply:
See the latest code, I set to 10.